File: src/main/java/graveler/SchedulerStorage.java
package graveler;

import net.minecraft.nbt.INBT;
import net.minecraft.util.Direction;
import net.minecraftforge.common.capabilities.Capability;
import net.minecraftforge.common.capabilities.Capability.IStorage;

public class SchedulerStorage implements IStorage<Scheduler> {
  @Override
  public INBT writeNBT(Capability<Scheduler> capability, Scheduler instance, Direction side) {
    return null;
  }

  @Override
  public void readNBT(
      Capability<Scheduler> capability, Scheduler instance, Direction side, INBT nbt) {
    // Do nothing
  }
}

File: src/main/kotlin/graveler/SchedulerStorage.kt
package graveler

import net.minecraft.nbt.CompoundNBT
import net.minecraft.nbt.INBT
import net.minecraft.util.Direction
import net.minecraft.util.ResourceLocation
import net.minecraftforge.common.capabilities.Capability
import net.minecraftforge.common.capabilities.Capability.IStorage

class SchedulerStorage : IStorage<Scheduler> {
  override fun writeNBT(
    capability: Capability<Scheduler>,
    instance: Scheduler,
    side: Direction?
  ): INBT? = CompoundNBT()

  override fun readNBT(
    capability: Capability<Scheduler>,
    instance: Scheduler,
    side: Direction?,
    nbt: INBT?
  ) {
    // Do nothing
  }

  companion object {
    val Resource = ResourceLocation(GravelerMod.ModId, "physics")
  }
}
